app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -671,11 +669,15 @@ void OpenXRInputSource::EmulateControllerFromHand(device::RenderMode renderMode,
         indexPinching = mSmoothIndexThumbDistance < kPinchThreshold;
     }
     delegate.SetPinchFactor(mIndex, pinchFactor);
-    bool triggerButtonPressed = indexPinching && !leftPalmFacesHead && mHasAimState;
+    bool triggerButtonPressed = indexPinching && !palmFacesHead && mHasAimState;
     delegate.SetButtonState(mIndex, ControllerDelegate::BUTTON_TRIGGER,
                             device::kImmersiveButtonTrigger, triggerButtonPressed,
                             triggerButtonPressed, 1.0);
-    if (leftPalmFacesHead) {
+    if (palmFacesHead) {
+#if defined(OCULUSVR)
+        if (mHandeness == Right)
+            return;
+#endif
```

Comment:
Hmm I have doubts about this, why would we want to return here?

Why not doing if palm faces head and is left hand?

There is an early return after this if-else block that will make oculus early return anyway (because in Oculus we don't have aim data if hands face the head)

Reply:
Actually I think the condition could be `if (palmFacesHead && !systemTakesOverWhenHandsFacingHead)` meaning that we handle it only when the system does not

Reply:
yes, good idea.